scene/resources/tile_set.h

```diff
@@ -144,6 +151,10 @@ class TileSet : public Resource {
 	Array _tile_get_shapes(int p_id) const;
 	Array _get_tiles_ids() const;
 	void _decompose_convex_shape(Ref<Shape2D> p_shape);
+	List<Vector2> _autotile_get_subtile_candidates_for_bitmask(int p_id, uint16_t p_bitmask) const;
+
+	static uint32_t _count_bitmask_bits(uint32_t bitmask);
+	static uint32_t _score_bitmask_difference(uint32_t bitmask, uint32_t ref_bitmask);
```

Comment:
Why is this static and protected?

Reply:
Also
```suggestion
	static uint32_t _count_bitmask_bits(uint32_t p_bitmask);
	static uint32_t _score_bitmask_difference(uint32_t p_bitmask, uint32_t p_ref_bitmask);
```

Reply:
These functions don't depend on or modify anything from a class instance, so they don't have to access `this`. Does Godot have a style rule against using static member functions?

Reply:
Does this function really need to be in the header?

Reply:
No, and it originally wasn't part of the class at all, but I asked if doing so for helper functions was normal in Godot's codebase and was told that they're normally part of the relevant class, so I moved it in.

Reply:
Oh, I see, you didn't specify what kind they were so I assume the assumption was made that it wasn't a static function

Reply:
Ah, okay. So I take it I should move them back out of the class, then, right?

Reply:
See what KoBeWi says as well, though my vote would be yes

Reply:
We don't really have rules for static/non-static functions (I think), but IMO a function should be static if it's supposed to be used without the class instance. If it "requires" class instance, even if not accessing any members, there's no reason to make it static.

The codebase is somewhat consistent with this. For example the TileSet on master has many non-static helper functions (like `_get_square_corner_or_side_terrain_peering_bit_polygon()`).

Reply:
Got it, doing non-static member functions to match that.